### ISSUE TITLE
Failed upload not reflected in exit code

### ISSUE THREAD

**Issue by kaos** [pypa/setuptools#869]
```
$ python setup.py sdist upload
  ...
running upload
Submitting dist/xxx-1.0.21.post22.tar.gz to https://pypi.example.com
Upload failed (404): Not Found
$ echo $?
0
```

**Comment by jaraco** [comment]
I suspect this is an inevitable outcome of the design of distutils. Can you investigate and see if there's a better behavior that could be adopted?

**Comment by kaos** [comment]
If I find the time I'd love to. Will see what I can do.
fre 9 dec. 2016 kl. 04:21 skrev Jason R. Coombs <notifications@github.com>:

**Comment by variable** [comment]
Having the same problem guys.

**Comment by variable** [comment]
My computer that returns the correct exit code has Python/Distutil version 2.7.10
The computer that suffers the problem has Python/Distutil version 2.7.6

**Comment by jaraco** [comment]
Indeed, when I do a diff, I see that the issue was in fact fixed in Python 2.7.7 per [python 21722](http://bugs.python.org/issue21722). There's nothing to be done here.

**Comment by variable** [comment]
I confirm that after upgrading to 2.7.12 worked.
